pkg/client/client.go

```diff
@@ -392,6 +392,18 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 					cluster.NumEndpoints, cluster.NumIdentities, cluster.NumSharedServices,
 					cluster.NumFailures, timeSince(time.Time(cluster.LastFailure)))
 				fmt.Fprintf(w, "   └  %s\n", cluster.Status)
+
+				fmt.Fprint(w, "   └  remote configuration: ")
+				if cluster.Config != nil {
+					fmt.Fprintf(w, "expected=%t, retrieved=%t", cluster.Config.Required, cluster.Config.Retrieved)
+					if cluster.Config.Retrieved {
+						fmt.Fprintf(w, ", cluster-id=%d, kvstoremesh=%t, sync-canaries=%t",
+							cluster.Config.ClusterID, cluster.Config.Kvstoremesh, cluster.Config.SyncCanaries)
+					}
```

Comment:
Is it hard to generate this output from the cluster-config itself? Otherwise, everytime we expose a new config, we need to edit here manually.

Reply:
🤔 I see your point, and I'm not an expert of the API definition part, but AFAIU this is not possible because we need to update the `openapi.yaml` definitions to include all the properties to be propagated. Hence, explicitly requiring to manually extend the definition and the usages in case a new entry gets added to the cluster config.

Reply:
I think that maybe what Yutaro is suggesting is that the details of how to serialize the cluster-config option should be defined in the cluster-config object, then this call should just call out to the stringify method on that object to get the canonical string representation?

Reply:
The tricky part (if I'm not missing anything) is that the cluster-config struct here is the one generated from the openapi definition, which is different from the one retrieved from etcd. Requiring to manually propagate the fields from one to another, and print them here. We could move the serialization function to somewhere else, but it would probably not help much (and it could not be a method of that object as autogenerated).